### ISSUE TITLE
I split my dataset into 10 classes, and I set init cls to 10, again reporting IndexError: Target 853 is out of bounds. How should this kind of mistake be solved

### ISSUE THREAD

**Issue by BLACKJACK0711** [xfflzl/CIM-CIL#6]

**Comment by xfflzl** [comment]
Hi, could you please share your folder architecture of your dataset?

**Comment by BLACKJACK0711** [comment]
> Hi, could you please share your folder architecture of your dataset?

Thank you for your concern about this issue. Please check your email for relevant introduction

**Comment by xfflzl** [comment]
Hi, thanks for your introduction! The datasets used in this repo are image datasets and the data folders of them are organized like following.
```
dataset main folder
├── train
    ├── class 0
        ├── class 0 train image 0
        ├── class 0 train image 1
        ├── ...
    ├── class 1
        ├── class 1 train image 0
        ├── class 1 train image 1
        ├── ...
    ├── ...
├── test
    ├── class 0
        ├── class 0 test image 0
        ├── ...
    ├── class 1
        ├── class 1 test image 0
        ├── ...
    ├── ...
```
With such folder architecture, the data files will be correctly read by `torchvision.datasets.ImageFolder`. You can find details from dataset loading classes called `iImageNet1000`, `iImageNet100` and `iVGGFace200` in `utils/data.py`.

To run this repo, please make sure your dataset samples are images and they are put in correct folders.

**Comment by BLACKJACK0711** [comment]
I converted my data into images through code, I am not sure what is the reason for the error，I added the print function below the logits line to see what its output looks like：

Its result is this：

Finally, please take a look at my specific error：





&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "xfflzl/CIM-CIL"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月20日(星期二) 中午11:05
***@***.***&gt;;
抄送:&nbsp;"BLACK ***@***.******@***.***&gt;;
主题:&nbsp;Re: [xfflzl/CIM-CIL] I split my dataset into 10 classes, and I set init cls to 10, again reporting IndexError: Target 853 is out of bounds. How should this kind of mistake be solved (Issue #6)





 
Hi, thanks for your introduction! The datasets used in this repo are image datasets and the data folders of them are organized like following.
 dataset main folder ├── train     ├── class 0         ├── class 0 train image 0         ├── class 0 train image 1         ├── ...     ├── class 1         ├── class 1 train image 0         ├── class 1 train image 1         ├── ...     ├── ... ├── test     ├── class 0         ├── class 0 test image 0         ├── ...     ├── class 1         ├── class 1 test image 0         ├── ...     ├── ...  
With such folder architecture, the data files will be correctly read by torchvision.datasets.ImageFolder. You can find details from dataset loading classes called iImageNet1000, iImageNet100 and iVGGFace200 in utils/data.py.
 
To run this repo, please make sure your dataset samples are images and they are put in correct folders.

**Comment by xfflzl** [comment]
Hi, it seems that the images are removed in the email.
